CLAUDE.md

```diff
@@ -0,0 +1,80 @@
+# Flight Control - Claude Development Guide
+
+## Common Bash Commands
+
+- `make build` - Build the project
+- `make unit-test` - Run unit tests (requires gotestsum)
+- `make generate` - Generate API code and mocks (requires mockgen)
+- `make deploy` - Deploy service locally in kind
+- `make deploy DB_VERSION=small` - Deploy with small DB (1000 devices max)
+- `make agent-vm` - Create agent VM for testing
+- `make agent-container` - Create agent container for testing
+- `make deploy-e2e-extras` - Start observability stack (Prometheus on :9090)
+- `bin/devicesimulator --count=100` - Simulate 100 devices
+
+## Core Files and Utility Functions
+
+- `cmd/flightctl/main.go` - Main CLI entry point
+- `cmd/flightctl-api/main.go` - API server entry point
+- `cmd/flightctl-agent/main.go` - Agent entry point
+- `internal/service/` - Core service logic
+- `internal/agent/` - Agent implementation
+- `api/v1alpha1/` - API definitions and types
+- `docs/developer/` - Developer documentation
+- `docs/user/` - User documentation
+
+## Code Style Guidelines
+
+- Use Go 1.23+ syntax and conventions
+- Follow Kubernetes-like API patterns
+- Use mTLS for agent communication, JWT for user authentication
+- Implement proper error handling with `internal/flterrors/`
+- Use structured logging with `pkg/log/`
+- Write tests for new functionality
+- Use mockgen for generating mocks
+
+## Testing Instructions
```

Comment:
make integration-test

It should use unit tests where possible.  Integration tests add a real DB, redis, and alertmanager - useful for testing how code interacts with those real systems.